### ISSUE TITLE
Postpone check of projects with blacklisted backend till there is no project in rate limit queue

### ISSUE THREAD

**Issue by Zlopez** [fedora-infra/anitya#888]
# Description
This is just a small enhancement that will make statistics much cleaner. Right now the projects are added to check queue after some time even if the project belongs to blacklisted backend. This enhancement propose to change the behavior to not include any project with blacklisted backend till all the projects, which were blocked by rate limit, were processed.

## Requirements
* Skip projects that are ready to check, but are part of the blacklisted backend, than we could remove the backend from blacklist as it is done right now. This way if we hit rate limit, it will be only with the projects that were in the rate limit queue before.

## How this will make Anitya better
This will make any statictics much cleaner and the check will be much faster, because it will skip plenty of projects that will be skipped anyway.
